src/test/java/AuthenticatedHttpServletTest.java

```diff
@@ -0,0 +1,172 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.sps.model.AuthenticatedHttpServlet;
+import com.google.sps.model.AuthenticationVerifier;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.GeneralSecurityException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/** Tests the AuthenticatedHttpServlet abstract class */
+@RunWith(JUnit4.class)
+public class AuthenticatedHttpServletTest {
+  // Mock abstract servlet with real constructor and real methods
+  private static final AuthenticationVerifier authVerifier =
+      Mockito.mock(AuthenticationVerifier.class);
+  private static final AuthenticatedHttpServlet servlet =
+      Mockito.mock(
+          AuthenticatedHttpServlet.class,
+          Mockito.withSettings()
+              .useConstructor(authVerifier)
+              .defaultAnswer(Mockito.CALLS_REAL_METHODS));
+
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+
+  private static final Boolean AUTHENTICATION_VERIFIED = true;
+  private static final Boolean AUTHENTICATION_NOT_VERIFIED = false;
+
+  private static final String ID_TOKEN_KEY = "idToken";
+  private static final String ID_TOKEN_VALUE = "sampleId";
+  private static final String ACCESS_TOKEN_KEY = "accessToken";
+  private static final String ACCESS_TOKEN_VALUE = "sampleAccessToken";
+
+  private static final Cookie sampleIdTokenCookie = new Cookie(ID_TOKEN_KEY, ID_TOKEN_VALUE);
+  private static final Cookie sampleAccessTokenCookie =
+      new Cookie(ACCESS_TOKEN_KEY, ACCESS_TOKEN_VALUE);
+  private static final Cookie emptyIdTokenCookie = new Cookie(ID_TOKEN_KEY, "");
+
+  private static final Cookie[] noCookies = new Cookie[] {};
+  private static final Cookie[] noAccessTokenCookies = new Cookie[] {sampleIdTokenCookie};
+  private static final Cookie[] noIdTokenCookies = new Cookie[] {sampleAccessTokenCookie};
+  private static final Cookie[] emptyIdTokenCookies =
+      new Cookie[] {emptyIdTokenCookie, sampleAccessTokenCookie};
+  private static final Cookie[] validCookies =
+      new Cookie[] {sampleIdTokenCookie, sampleAccessTokenCookie};
+
+  @Before
+  public void init() throws IOException {
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
+  }
+
+  @After
+  public void clear() {
```

Comment:
Prefer to use `tearDown` for the method annotated with `@After`. It's a standard naming convention in JUnit and related frameworks.

Also, you don't need to clear the stringWriter, because it's an instance variable, and JUnit will create a new instance of your class to run each test method.

Reply:
Removed this block, but noted!